artemis-server/src/main/java/org/apache/activemq/artemis/core/remoting/server/impl/RemotingServiceImpl.java

```diff
@@ -582,7 +582,7 @@ public void connectionCreated(final ActiveMQComponent component,
    public void addConnectionEntry(Connection connection, ConnectionEntry entry) {
       connections.put(connection.getID(), entry);
       if (AuditLogger.isResourceLoggingEnabled()) {
-         AuditLogger.createdConnection(connection.getProtocolConnection().getProtocolName(), connection.getID(), connection.getRemoteAddress());
+         AuditLogger.createdConnection(connection.getProtocolConnection() == null ? null : connection.getProtocolConnection().getProtocolName(), connection.getID(), connection.getRemoteAddress());
```

Comment:
Having the local-address in the actual audit log also seems like it would be [more] useful here, as in the handshake failure case. Probably a separate change though.

Reply:
I agree on both counts.